scanpipe/pipes/flag.py

```diff
@@ -102,6 +105,20 @@ def flag_ignored_patterns(project, patterns):
     return update_count
 
 
+def flag_and_ignore_files_over_max_size(resource_qs, file_size_limit):
+    """
+    Flag codebase resources which are over the max file size for scanning
+    and return all other files within the file size limit.
+    """
+    if not file_size_limit:
+        return resource_qs
+
+    resource_qs.filter(size__gte=file_size_limit).update(
+        status=IGNORED_BY_MAX_FILE_SIZE
+    )
+    return resource_qs.filter(~Q(status=IGNORED_BY_MAX_FILE_SIZE))
```

Comment:
This is not consistent with the existing flag method. It should return the update count.
The filtering should be done outside of the flag function